api/lib/domain/services/user-reconciliation-service.js

```diff
@@ -38,12 +53,36 @@ async function findMatchingSchoolingRegistrationIdForGivenOrganizationIdAndUser(
     throw new NotFoundError('There were no schoolingRegistrations matching with names');
   }
 
-  const matchingSchoolingRegistration = _.find(schoolingRegistrations, { 'id': schoolingRegistrationId });
-  if (!_.isNil(matchingSchoolingRegistration.userId)) {
-    throw new SchoolingRegistrationAlreadyLinkedToUserError();
+  const matchedSchoolingRegistration = _.find(schoolingRegistrations, { 'id': schoolingRegistrationId });
+  await checkIfStudentIsAlreadyReconciledOnTheSameOrganization(matchedSchoolingRegistration, userRepository);
+
+  return matchedSchoolingRegistration;
+}
+
+async function checkIfStudentIsAlreadyReconciledOnTheSameOrganization(matchingSchoolingRegistration, userRepository) {
+  if (!_.isNil(matchingSchoolingRegistration.userId))  {
+    const userId = matchingSchoolingRegistration.userId ;
+    const user = await userRepository.getUserAuthenticationMethods(userId);
+    const authentificationMethod = getUserAuthenticationMethodWithObfuscation(user);
+
+    const detail = 'Un compte existe déjà pour l\'élève dans le même établissement.';
+    const error = STUDENT_RECONCILIATION_ERRORS.IN_SAME_ORGANIZATION[authentificationMethod.authenticatedBy];
+    const meta = { displayShortCode: error.displayShortCode, value: authentificationMethod.value };
```

Comment:
Je ne suis pas sûr de la nécessité d'introduire un champ `meta` pour renvoyer un code d'erreur fonctionnel et de se baser dessus côté front pour savoir quel message d'erreur afficher. Je pense qu'il serait mieux, et suffisant, de renvoyer dans l'attribut `code` de l'objet `error` les valeurs `ACCOUNT_WITH_EMAIL_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION`, `ACCOUNT_WITH_USERNAME_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION`... qui correspondent aux clés de traduction et représentent des codes d'erreurs techniques. La documentation de JSON:API nous dit d'ailleurs très bien que l'attribut `code` sert à ça:

> code: an application-specific error code, expressed as a string value.

Les codes R11, R12... ne sont que des codes définis par le métier et ne devraient pas être présents dans le code de l'API selon moi.

Reply:
Selon la spec json-api, meta: a meta object containing non-standard meta-information about the error.
Dans ce usecase, il permet d'envoyer à notre IHM le résultat de l'obfuscation des methodes d'authentification de l'utilisateur. Cette donnée n'est pas présente coté client , c'est l'api qui la renvoie.
En ce qui concerne le shortCode, pour ce usecase précis, je le vois comme un code de règle de gestion de l'api compréhensible par le dev, le métier et l'utilisateur final. Du coup ca me choque pas de l'avoir dans notre base de code. 
Vu au point tech, on essaye de creuser avec le métier pour trancher.

Reply:
Vu ensemble, on garde le shortCode dans la base de code.